internal/k8shandler/collection.go

```diff
@@ -31,107 +31,121 @@ func (clusterRequest *ClusterLoggingRequest) CreateOrUpdateCollection() (err err
 	cluster := clusterRequest.Cluster
 	collectorConfig := ""
 	collectorConfHash := ""
+
+	// Default to vector collector type
+	collectionSpec := &logging.CollectionSpec{
+		Type: logging.LogCollectionTypeVector,
+	}
 	log.V(9).Info("Entering CreateOrUpdateCollection")
-	log.V(3).Info("creating collector using", "spec", cluster.Spec.Collection)
 	defer func() {
 		log.V(9).Info("Leaving CreateOrUpdateCollection")
 	}()
 
-	// there is no easier way to check this in golang without writing a helper function
-	// TODO: write a helper function to validate Type is a valid option for common setup or tear down
-	if cluster.Spec.Collection != nil && cluster.Spec.Collection.Type.IsSupportedCollector() {
-
-		var collectorType = cluster.Spec.Collection.Type
-
-		//TODO: Remove me once fully migrated to new collector naming
-		if err = clusterRequest.removeCollector(constants.FluentdName); err != nil {
-			log.V(2).Info("Error removing legacy fluentd collector.  ", "err", err)
+	if cluster != nil && clusterRequest.Forwarder.Name == constants.SingletonName {
+		if !clusterRequest.isManaged() {
+			return nil
 		}
-
-		// LOG-2620: containers violate PodSecurity
-		if err = clusterRequest.addSecurityLabelsToNamespace(); err != nil {
-			log.Error(err, "Error adding labels to logging Namespace")
-			return
-		}
-
-		if err = collector.ReconcileServiceAccount(clusterRequest.EventRecorder, clusterRequest.Client, cluster.Namespace, constants.CollectorServiceAccountName, utils.AsOwner(cluster)); err != nil {
-			log.V(9).Error(err, "collector.ReconcileServiceAccount")
-			return
+		if cluster.Spec.Collection != nil && cluster.Spec.Collection.Type.IsSupportedCollector() {
+			// Change collector type dependent on clusterLogging
+			collectionSpec = cluster.Spec.Collection
+		} else {
+			if err = clusterRequest.RemoveServiceAccount(); err != nil {
+				return
+			}
+			if err = clusterRequest.removeCollector(); err != nil {
+				return
+			}
 		}
-		if err = collector.ReconcileRBAC(clusterRequest.Client, cluster.Namespace, constants.CollectorServiceAccountName, utils.AsOwner(cluster)); err != nil {
-			log.V(9).Error(err, "collector.ReconcileRBAC")
-			return
+	}
+	//TODO: Remove me once fully migrated to new collector naming
```

Comment:
You can remove this logic for deleting the old collector name that was introduced in 5.4(EOL)